webapp/tests/karma/ts/services/xml-forms.service.spec.ts

```diff
@@ -777,6 +777,43 @@ describe('XmlForms service', () => {
 
     });
 
+    it('does not return a form with a truthy expression if the user does not have relevant permissions', () => {
```

Comment:
Please add a unit test where a form has a permission and an expression and: 
1. user has permission and expression is valid
2. user doesn't have permission and expression is valid
3. user has permission and expression is invalid.